### DISCUSSION TITLE
Device control plane event configuration

### DISCUSSION THREAD

**Discussion by pranavkumaradarji-eaton** [edgexfoundry/community#267] [Q&A]
Our project’s codebase is using EdgeX version 2.3. We are currently testing control plane events for device CRUD operations. We attempted to add a new device and listened on the MQTT topic ‘edgex/system-events/#,’ but we didn’t observe any events being generated.

**Question:** Is there any configuration tweak required to enable device control plane event functionality?

**Comment by cloudxxx8** [comment]
The system event will be sent to Redis pub/sub by default.  Have you configured core-metadata to use MQTT message bus?

**Reply by pranavkumaradarji-eaton** [reply]
Yes, Core-metadata is configured to use "MQTT" message bus, and also changed PublishTopicPrefix = "edgex/system-events" , still events aren't received. @cloudxxx8

**Reply by cloudxxx8** [reply]
OK, it should work by design and implementation.  We may see if there is anyone from the community can help on this.

**Comment by M0hanrajp** [comment]
Hello @pranavkumaradarji-eaton, 

Could you try checking what error might be posted by the logs by referring the doc [EdgeX Foundry Container Logs
](https://docs.edgexfoundry.org/3.0/getting-started/Ch-GettingStartedDockerUsers/#edgex-foundry-container-logs)?

I also faced a similar issue, to send the EdgeX data to a public MQTT broker hosted by HiveMQ. You can then watch\
for the EdgeX event data via HiveMQ provided MQTT browser client. Here I was not able to receive any events over\
HiveMQ. I was running the example from the doc [Exporting Data
](https://docs.edgexfoundry.org/3.0/getting-started/quick-start/#exporting-data)

By using the logs generated from the core services I think I ran into BROKER_ADDRESS being not identified.

Still not fixed my issue, but you can follow the below documentation for more info of your error.

Thanks.

**Comment by cloudxxx8** [comment]
does your codebase contain this PR?
https://github.com/edgexfoundry/edgex-go/pull/4101/files
we tested and it worked before
